File: rb/features.rb
# BbcWeb SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module BbcWebFeatures
  def self.make_feature(name)
    case name
    when "base"
      BbcWebBaseFeature.new
    when "test"
      BbcWebTestFeature.new
    else
      BbcWebBaseFeature.new
    end
  end
end

File: py/features.py
# BbcWeb SDK feature factory

from feature.base_feature import BbcWebBaseFeature
from feature.test_feature import BbcWebTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BbcWebBaseFeature(),
        "test": lambda: BbcWebTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
